migrations/20240214120130_add_is_anonymous_column.up.sql

```diff
@@ -0,0 +1,13 @@
+do $$
+begin
+   alter table {{ index .Options "Namespace" }}.users 
```

Comment:
Not sure why the column is generated? A user without an email address is not an anonymous user.

An anonymous user is that which used `signInAnonymously()`. So it shouldn't be a generated column, but rather one that's settable.

Reply:
Basically the first time you call `linkIdentity` on an anonymous user, set `is_anonymous` to false.

Reply:
that was the definition we agreed on internally

it also completely rules out any data integrity scenarios where a user with an email / phone has `is_anonymous` set to true by accidentally

a user is only considered an anonymous user if it doesn't have an email _and_ phone number